src/main/java/com/phonebookservice/controller/AbstractController.java

```diff
@@ -0,0 +1,33 @@
+package main.java.com.phonebookservice.controller;
+
+import main.java.com.phonebookservice.model.Model;
+import main.java.com.phonebookservice.server.DBIf;
+
+public abstract class AbstractController {
+    private DBIf databse = null;
```

Comment:
This should be final. Also don't initialize variable outside constructor except constants. 

```
 final private IDB databse;
```

Reply:
should be protected not private.
Now I got your point, why did u have getDB. If you are going to use the instance variable in the child class, make it protected in the parent class. 

please write down here the difference between public vs protected vs private